syntax = "proto3";
package protos.playbooks;

import "protos/base.proto";
import "google/protobuf/wrappers.proto";


message ServiceMetricExecutionTask {
  google.protobuf.StringValue service_name = 1;
  google.protobuf.StringValue environment_name = 2;
  google.protobuf.StringValue metric_family = 3;
  google.protobuf.StringValue metric = 4;
  google.protobuf.StringValue process_function = 5;
}

message QueryMetricExecutionTask {
  repeated string queries = 1;
  google.protobuf.StringValue formula = 2;
  google.protobuf.StringValue process_function = 3;
}

message PlaybookDatadogTask {
  enum TaskType {
    UNKNOWN = 0;
    SERVICE_METRIC_EXECUTION = 1;
    QUERY_METRIC_EXECUTION = 2;
  }

  TaskType type = 1;
  oneof task {
    ServiceMetricExecutionTask service_metric_execution_task = 2;
    QueryMetricExecutionTask query_metric_execution_task = 3;
  }
}